HotFix.Specification/Specification.cs

```diff
@@ -64,13 +59,14 @@ public Specification Run()
             {
                 if (Exception != null)
                 {
+                    if (exception.GetType() == typeof(DelightfullySuccessfulException))
+                        throw new AssertFailedException("The expected exception type was not thrwon. All instructions successfully executed instead");
+
                     if (Exception != exception.GetType())
-                        throw new AssertFailedException("The expected exception type was not thrown");
+                        throw new AssertFailedException($"The expected exception type was not thrown. The type '{exception.GetType()}' was thrown instead");
```

Comment:
`throw new AssertFailedException($"The test expected an exception of type {Exception.FullName} but an exception of type {exception.GetType().FullName} was thrown instead");`

I think at this point we only care about the exception type/name so we want to print the *namespace and name*, aka `System.IO.IOException` etc